assets/scripts/dialogs/GeotagDialog.jsx

```diff
@@ -48,148 +44,138 @@ L.Icon.Default.mergeOptions({
+    label = props.addressInformation.label
+    // If there's no prior location data, use the user's location, if available
+    // In this case, display the map view, but no marker or popup
+  } else if (props.userLocation) {
+    mapCenter = {
+      lat: props.userLocation.latitude,
+      lng: props.userLocation.longitude
+    }
+    zoom = MAP_LOCATION_ZOOM
+    // As a last resort, show an overview of the world.
+  } else {
+    mapCenter = DEFAULT_MAP_LOCATION
+    zoom = DEFAULT_MAP_ZOOM
   }
 
-  static defaultProps = {
-    dpi: 1.0
+  return {
+    mapCenter,
+    zoom,
+    markerLocation,
+    label
   }
+}
 
-  constructor (props) {
-    super(props)
-
-    // Determine initial map center, and what to display
-    let mapCenter, zoom, markerLocation, label
-
-    // If street has a location object, use its position and data
-    if (props.street.location) {
-      mapCenter = props.street.location.latlng
-      zoom = MAP_LOCATION_ZOOM
-      markerLocation = props.street.location.latlng
-      label = props.street.location.label
-      // If we've previously saved marker position, re-use that information
-    } else if (props.markerLocation) {
-      mapCenter = props.markerLocation
-      zoom = MAP_LOCATION_ZOOM
-      markerLocation = props.markerLocation
-      label = props.addressInformation.label
-      // If there's no prior location data, use the user's location, if available
-      // In this case, display the map view, but no marker or popup
-    } else if (props.userLocation) {
-      mapCenter = {
-        lat: props.userLocation.latitude,
-        lng: props.userLocation.longitude
-      }
-      zoom = MAP_LOCATION_ZOOM
-      // As a last resort, show an overview of the world.
-    } else {
-      mapCenter = DEFAULT_MAP_LOCATION
-      zoom = DEFAULT_MAP_ZOOM
-    }
+/*
+This Dialog uses the generic Dialog component and combines it with the GeoSearch
+and LocationPopup components as well as a map to display the coordinates on
+It handles:
+setting, displaying, and clearing location information associated with a 'street'
+user permission checks for whether location can be updated for a street based on the user
+reverse geocodeing based on user input (the user can click on the map to reverse geocode at the
+  clicked point, or it can drag the marker to a new location to confirm)
 
-    this.state = {
-      mapCenter: mapCenter,
-      zoom: zoom,
-      renderPopup: !!markerLocation,
-      markerLocation: markerLocation,
-      label: label,
-      bbox: null,
-      geocodeAvailable: !!PELIAS_API_KEY
-    }
+It is tested primary via cypress at the moment
+ */
+function GeotagDialog () {
+  const props = {
+    street: useSelector((state) => state.street),
+    markerLocation: useSelector((state) => state.map.markerLocation),
+    addressInformation: useSelector((state) => state.map.addressInformation),
+    userLocation: useSelector((state) => state.user.geolocation.data)
```

Comment:
I'm wondering if it's possible to refactor this `props` object out? It was a temporary measure to get state from Redux store without having to change instances of `prop.*` anywhere else in the component yet. This particular construction isn't used anywhere else in our codebase so it will look strange the next time we look at this, and it also requires that we keep the `.propTypes` defined because we use the `props` variable still. Usually when we pull state from Redux, and no longer have real props, we are fine with removing the PropTypes.

This can raise the question of whether we care about type safety, but that's definitely a much longer conversation for another time.

Reply:
Yes, this sounds good to me! I wanted confirmation that it wasn't going to mess anything up or be inconsistent with the rest of the codebase.

Reply:
@louh I have some thoughts on this, but not a clear path on how to refactor it out. The tricky part still is the sort of conditional way we're pulling out difference pieces of state from Redux based on what exists. It might depend on some of the above questions. I'm also open to it being technical debt for now until there's a need to revisit the component again.